### PR TITLE
Added R 3.4.0

### PR DESCRIPTION
**Pull request by JusticeForMikeBrown**
Hi

Maybe I need to delete my repository.  I have three local branches but I cannot figure out how to get rid of my commit history.  I reset head back to upstream but then I am not able to push to my remote branches without merging from my origin which again has the stuff I do not want.   Noob error, obviously.

Anyway, I have added R 3.4.0.  It was built successfully on Centos 6.8 and 7.3

### CODE REVIEW

File: var/spack/repos/builtin/packages/r/package.py [hunk 1]
```diff
@@ -36,10 +36,11 @@ class R(AutotoolsPackage):
     Please consult the R project homepage for further information."""
 
     homepage = "https://www.r-project.org"
-    url = "https://cloud.r-project.org/src/base/R-3/R-3.3.2.tar.gz"
+    url = "https://cran.r-project.org/src/base/R-3/R-3.4.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r/package.py]: I would switch back to `cloud`. See #2956.

**Comment by davydden** [comment]
> I have three local branches but I cannot figure out how to get rid of my commit history

i find atlassian (bitbucket) guides usefull https://www.atlassian.com/git/tutorials/syncing

**Comment by adamjstewart** [comment]
You'll probably have to force pull upstream develop from your local repository and then force push it to origin develop to overwrite those changes. Just make sure you don't have any local changes that will be lost.

**Comment by adamjstewart** [comment]
@hartzell You may be interested in this.

**Comment by JusticeForMikeBrown** [comment]
@adamjstewart 

Let me know if I need to make any changes per the CI check.

**Comment by adamjstewart** [comment]
There's a problem with the latest Sphinx release. See #4243. I'm looking into it now.

**Comment by JusticeForMikeBrown** [comment]
@adamjstewart 

Thanks!  My develop branch is now synced!

user@node103[~/repos/spack]$ git remote -v

```
origin  git@github.com:JusticeForMikeBrown/spack.git (fetch)
origin  git@github.com:JusticeForMikeBrown/spack.git (push)
upstream        git@github.com:LLNL/spack.git (fetch)
upstream        git@github.com:LLNL/spack.git (push)

git reset --hard  upstream/develop
git push -f -u origin develop
```

**Comment by hartzell** [comment]
I'm confused about all of the commits to add 1 line to the R package.  I'm guess that that's what the "Wow, is `git` every easy!" discussion is about.

Do we squash merges or will they all end up in the develop branch?  If we do not squash it might be nice to clean it up and push/force push new content to this branch before it gets merged.

**Comment by adamjstewart** [comment]
We squash all merges, so there's nothing to fear.

**Pull request closed without merging**